pkg/skaffold/debug/cnb.go

```diff
@@ -117,11 +118,18 @@ func adjustCommandLine(m cnb.BuildMetadata, ic imageConfiguration) (imageConfigu
 
 	for _, p := range m.Processes {
 		if p.Type == processType {
+			// Direct: p.Command is the command and p.Args are the arguments
 			if p.Direct {
-				// p.Command is the command and p.Args are the arguments
-				ic.arguments = append([]string{p.Command}, p.Args...)
-				return ic, func(transformed []string) []string {
-					return append([]string{"--"}, transformed...)
+				// Detect and unwrap `/bin/sh -c ...`-style command lines; GCP Buildpacks turn Procfiles into `/bin/bash -c ...`
+				if (p.Command == "/bin/sh" || p.Command == "/bin/bash") && len(p.Args) >= 2 && p.Args[0] == "-c" {
+					p.Command = p.Args[1]
```

Comment:
is the command is `Command: "/bin/sh", Args: []string{"-c", "command arg1 arg2"}` as in your test case, then 
`p.Command = strings.Split(p.Args[2], " ")[0]` and
`p.Args = strings.Split(p.Args[2], " ")[1:]`

or is it the case at this point, p.Args is already a space separated array of all args?

Reply:
So `p.Args[1]` is the script provided to `sh -c`.  p.Args[2:] are then mapped to arguments for that shell script.  So it's very rarely done:
```console
$ sh -c 'echo "0 - $0"; i=0; for arg in $*; do i=$(($i + 1)); echo "$i - $arg"; done' hi there how are you
0 - hi
1 - there
2 - how
3 - are
4 - you
```

Note that `strings.Split(..., " ")` doesn't take into account quoted strings, etc.

Reply:
(It's entirely possible that they provide `/bin/sh -x -c '...'` but I'm fine with giving up and saying "🙅‍♂️ _no debug for you_".)

Reply:
lol.